tests/integration/test_playstation.py

```diff
@@ -0,0 +1,135 @@
+    PROJECT_ID = 42
+    playstation_dump = load_dump_file("playstation.prosperodmp")
+    mini_sentry.add_full_project_config(PROJECT_ID)
+    outcomes_consumer = outcomes_consumer()
+    relay = relay_with_processing(
+        {
+            "limits": {
+                "max_attachments_size": len(playstation_dump) - 1,
+            }
+        }
+    )
+
+    with pytest.raises(requests.exceptions.HTTPError) as exc_info:
+        _ = relay.send_playstation_request(PROJECT_ID, playstation_dump)
+
+    response = exc_info.value.response
+    assert response.status_code == 400, "Expected a 400 status code"
+
+
+# Turns out these are not data_dog metrics but sentry_metrics hence the entire thing is not working
+def test_playstation_with_feature_flag(
+    mini_sentry, relay_with_processing, outcomes_consumer
+):
+    PROJECT_ID = 42
+    playstation_dump = load_dump_file("playstation.prosperodmp")
+    mini_sentry.add_full_project_config(
+        PROJECT_ID,
+        extra={"config": {"features": ["projects:relay-playstation-ingestion"]}},
+    )
+    outcomes_consumer = outcomes_consumer()
+    relay = relay_with_processing()
+
+    response = relay.send_playstation_request(PROJECT_ID, playstation_dump)
+    assert response.ok
+
+    # Get these outcomes since the feature flag is not enabled:
+    outcomes = outcomes_consumer.get_outcomes()
+    assert len(outcomes) == 2
+    assert outcomes[0]["reason"] == "no_event_payload"
+    assert outcomes[1]["reason"] == "no_event_payload"
+
+
+def test_playstation_attachment(
+    mini_sentry, relay_with_processing, outcomes_consumer, attachments_consumer
+):
+    PROJECT_ID = 42
+    playstation_dump = load_dump_file("playstation.prosperodmp")
+    mini_sentry.add_full_project_config(
+        PROJECT_ID,
+        extra={"config": {"features": ["projects:relay-playstation-ingestion"]}},
+    )
+    outcomes_consumer = outcomes_consumer()
+    attachments_consumer = attachments_consumer()
+    relay = relay_with_processing()
+
+    bogus_error = {
+        "event_id": "cbf6960622e14a45abc1f03b2055b186",
+        "type": "error",
+        "exception": {"values": [{"type": "ValueError", "value": "Should not happen"}]},
+    }
+    envelope = Envelope()
+    envelope.add_event(bogus_error)
+
+    # Add the PlayStation dump as an attachment
+    envelope.add_item(
+        Item(
+            type="attachment",
+            payload=PayloadRef(bytes=playstation_dump),
+            headers={
+                "attachment_type": "playstation.prosperodump",
+                "filename": "playstation.prosperodmp",
+                "content_type": "application/octet-stream",
+            },
+        )
+    )
+    relay.send_envelope(PROJECT_ID, envelope)
+
+    outcomes = outcomes_consumer.get_outcomes()
+    assert len(outcomes) == 0
+    # TODO: Add stronger assert later to check we make it into the expand function and add test
+    # to check that we don't make it into if the feature is not enabled.
```

Comment:
It'd be good if you setup the tests as a relay chain (usually a parametrized test), then we can also test that sending them through multiple layers works as expected.